pkg/commonvpc/vpc.go

```diff
@@ -46,7 +46,16 @@ func zoneFromVPCResource(r vpcmodel.VPCResourceIntf) (*Zone, error) {
 type NetworkInterface struct {
 	vpcmodel.VPCResource
 	vpcmodel.InternalNode
-	Vsi string `json:"-"`
+	Vsi               string `json:"-"`
+	numberOfNifsInVsi int
```

Comment:
How about storing just a pointer to the parent `commonvpc.Vsi` instead of these two fields?
You can then use `parentVSI.Name` and `len(parentVsi.VPCnodes)` to obtain these two values.

Reply:
a lot of tests use vsi name but does not init the vsi struct, that need a big change